src/jinacld_tools/aws/services/cloudwatch.py

```diff
@@ -22,67 +20,37 @@ def __init__(self):
         self.logger = get_logger(self.__class__.__name__)
         self._client_wrapper = AWSClientWrapper(service='cloudwatch')
         self._client = self._client_wrapper.client
-        self.metrics = "" # url of metrics collected
 
-    def fetch(self, ec2ARN, endTime, metrics):
+    def fetch(self, ec2_arn, end_time, metrics):
         '''
             given an ec2 instance's ARN, e.g. "arn:aws:ec2:region:accountID:instanceID
             collect system metrics from (launch time to user defined end time)
             - AWS provided : CPU, Disk, Network data, status check(VM instance and its hardware)
             - custom : RAM
         '''
-        _, _, _, region, _, instanceID = ec2ARN.split(":")
-        '''
-        from boto3.ec2.connection import EC2Connection
-        conn = EC2Connection(ACCESS_KEY_ID, SECRET_ACCESS_KEY)
-        conn = EC2Connection()
-        print(f'conn: {conn}')'''
-
-        #session = boto3.Session(profile_name="dev", region_name="us-east-1")
-        '''
-        ec2 = boto3.client('ec2', region_name="us-east-2", #aws_instance_id=
-                           aws_access_key_id=ACCESS_KEY_ID, aws_secret_access_key=SECRET_ACCESS_KEY)
-        for each in ec2.describe_instances()["Reservations"]:
-            print(each)
-              #list_bucket_metrics_configurations())'''
-
-        ec2 = boto3.resource('ec2', region_name=region,
-                aws_access_key_id=ACCESS_KEY_ID, aws_secret_access_key=SECRET_ACCESS_KEY)
-
-        instance = ec2.Instance(instanceID)
-
-        for instance in ec2.instances.filter(Filters=[{'Name': 'instance-id', 'Values':[instanceID]}]):
-            print(instance)
-            print(instance.state)
-            #print(instance.report_status())
-
-        metrics = {}
-
-        '''
-        metric_names = ['DiskReadOps', 'DiskWriteBytes', CPUUtilization]
-        cloudwatch = boto3.resource('cloudwatch', region_name=region,
-                aws_access_key_id=ACCESS_KEY_ID, aws_secret_access_key=SECRET_ACCESS_KEY)
-        for metric_name in metric_names:
-            metrics[metric_name] = cloudwatch.metrics.filter(Namespace='AWS/EC2', MetricName=metric_name)
-        print(metrics)'''
+        _, _, _, region, _, instance_id = ec2_arn.split(":")
 
         cloudwatch = boto3.client('cloudwatch', region_name=region,
                 aws_access_key_id=ACCESS_KEY_ID, aws_secret_access_key=SECRET_ACCESS_KEY)
 
         responses = []
+        ec2 = boto3.resource('ec2', region_name=region,
```

Comment:
This should be done in the `ec2` service.

Reply:
With only client I haven't seen easy access to ec2 instance's launch_time